File: lib/features/home/data/repositories/home_repository_impl.dart
import 'package:islami_admin/features/home/data/datasources/home_remote_data_source.dart';
import 'package:islami_admin/features/home/domain/entities/home_stats.dart';
import 'package:islami_admin/features/home/domain/repositories/home_repository.dart';

class HomeRepositoryImpl implements HomeRepository {
  final HomeRemoteDataSource remoteDataSource;

  HomeRepositoryImpl({required this.remoteDataSource});

  @override
  Future<HomeStats> getHomeStats() {
    return remoteDataSource.getHomeStats();
  }
}

File: shared/src/commonMain/kotlin/com/islami/data/repository/HomeRepositoryImpl.kt
package com.islami.data.repository

import com.islami.core.error.Result
import com.islami.data.remote.firebase.FirebaseFirestoreClient
import com.islami.domain.entities.AppSettings
import com.islami.domain.entities.HomeStats
import com.islami.domain.repositories.HomeRepository

class HomeRepositoryImpl(
    private val firestoreClient: FirebaseFirestoreClient
) : HomeRepository {

    override suspend fun getHomeStats(): Result<HomeStats> {
        return firestoreClient.getDocument(
            collection = "stats",
            documentId = "home",
            serializer = HomeStats.serializer()
        )
    }

    override suspend fun getAppSettings(): Result<AppSettings> {
        return firestoreClient.getDocument(
            collection = "settings",
            documentId = "app",
            serializer = AppSettings.serializer()
        )
    }
}
